[DUOS-1818][risk=no] Remove additionalEmail, academicEmail, and checkNotifications

connorlbark opened the pull request:
## Addresses 

https://broadworkbench.atlassian.net/browse/DUOS-1818

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

rushtong commented on src/test/java/org/broadinstitute/consent/http/db/UserDAOTest.java:
```diff
@@ -169,22 +169,6 @@ public void testInsertDACUser() {
         // No-op ... tested in `createUser()`
     }
 
-    @Test
-    public void testUpdateDACUser_case1() {
-        User user = createUser();
-        Institution firstInstitute = createInstitution();
-        String newEmail = getRandomEmailAddress();
-        userDAO.updateUser(
-                "Dac User Test",
-                user.getUserId(),
-                newEmail,
-                firstInstitute.getId()
-                );
-        User user2 = userDAO.findUserById(user.getUserId());
-        assertEquals(user2.getAdditionalEmail(), newEmail);
-        assertEquals(user2.getInstitution().getId(), firstInstitute.getId());
-    }
```
I think we still need this test case,  but without the email component. This test is actually testing two cases, institution and email updates. We should maintain the institution part of this test. I do see usages in the helper class, but that's not sufficient - we should have a test for `userDAO.updateUser` here too.

connorlbark replied:
ahh, didn't see the institution change - brought the test back!

rushtong commented on src/main/java/org/broadinstitute/consent/http/service/EmailNotifierService.java:
```diff
@@ -263,19 +263,11 @@ public void sendDataCustodianApprovalMessage(String toAddress,
     }
 
     private Set<String> getEmails(List<User> users) {
```
Lines 151 and 163 also make reference to `additionalEmail` so we should refactor those a bit. It looks like we can simplify that quite a bit in both cases:
```
Set<String> emails = Set.of(data.get("email"));
```

rushtong commented on src/main/java/org/broadinstitute/consent/http/enumeration/UserFields.java:
```diff
@@ -4,7 +4,6 @@
 import java.util.List;
 
 public enum UserFields {
-  ACADEMIC_BUSINESS_EMAIL("academicEmail", true),
```
The ticket misses this, but I do see this in the code. We should deprecate this from `DataAccessRequestData` as well, my apologies for not covering that in DUOS-1818. You'll see some convoluted logic around this in `ElectionService` too. 

```
String researcherEmail = Objects.nonNull(researcher) ?
        researcher.getEmail() :
        Objects.nonNull(dar.getData().getAcademicEmail()) ?
                dar.getData().getAcademicEmail() :
                dar.getData().getResearcher();
```
should probably be:
```
String researcherEmail = Objects.nonNull(researcher) ? researcher.getEmail() : null;
```
with an additional null check around 
```
emailNotifierService.sendDataCustodianApprovalMessage(custodian.getEmail(), darCode, mailDTOS, custodian.getDisplayName(), researcherEmail);
```
so we're not trying to send an email to someplace we cannot send it. This would deserve a log.error call as well so we can catch these kinds of things.

connorlbark replied:
thanks for the tips! removed the field; seemed it wasn't used anywhere else.

quazi-broad reviewed (commented):
Looking good so far, when loading the changes and running in a local instance, everything compiled okay. When hitting the /user/me endpoint academicEmail and checkNotifications was being returned under "researcherProperties" - is that expected?

I did a broad search across the consent repo for all references to academicEmail, additionalEmail, and checkNotifications and found the following occurences:

DarService.scala
line 55: academicEmail

User.java
line 30: additional_email

InstititutionDAO.java
lines 64: additional_email
lines 68: additional_email

DacDAO.java
line 75: additional_email

JsonProtocols.scala
line 146: checkNotifications
line 120: academicEmail
line 213: academicEmail


DataAccessRequestData.java
line 29: add "additionalEmail" and "checkNotifications" to the list of DEPRECATED_PROPS?

ResearcherModels.scala
line 15: academicEmail
line 41: checkNotifications


DataAccessRequestModels.scala
line 65: academicEmail

rushtong commented:
We don't need to modify the scala classes for this PR - those are all optional and should remain so.

connorlbark commented:
thanks for all the great feedback! i believe i addressed everything: DAO calls updated, additionalEmail and checkNotifications added to deprecated fields, also fixed merge conflicts with master.

quazi-broad reviewed (approved):
Thanks for the changes! I think this is looking good.

rushtong commented on src/main/java/org/broadinstitute/consent/http/models/DataAccessRequestData.java:
```diff
@@ -26,7 +26,7 @@ public class DataAccessRequestData {
             "eraAuthorized", "nihUsername", "linkedIn", "orcid", "researcherGate", "datasetDetail",
             "datasets", "datasetId", "validRestriction", "restriction", "translatedUseRestriction",
-            "createDate", "sortDate");
+            "createDate", "sortDate", "additionalEmail", "checkNotifications" );
```
This is a non-issue, but I thought I would describe it for posterity. `DEPRECATED_PROPS` is used to create a simplified version of the DAR where it filters out all of these values. I initially thought we would still need to do that filtering, but it turns out that since you removed the getters/setters, these fields are never populated in any db call. So, nice work.

rushtong reviewed (approved):
Nice work, thank you!
